### PR TITLE
[FTX] Adding new exchange

### PR DESCRIPTION
**Pull request by makarid**

### CODE REVIEW

File: xchange-core/src/main/java/org/knowm/xchange/dto/Order.java [hunk 1]
```diff
@@ -431,6 +431,10 @@ public OrderType getOpposite() {
     REJECTED,
     /** Order has expired it's time to live or trading session and been removed from order book */
     EXPIRED,
+    /** Order is open and waiting to be filled */
+    OPEN,
+    /** Order has been either filled or cancelled */
+    CLOSED,
```
Comment (earce) [inline xchange-core/src/main/java/org/knowm/xchange/dto/Order.java:437]: Some of these states overlap each other, a partially filled order is technically opened, same a filled order can be closed, I am not sure if these states have not been added here before for a reason?

@timmolter

File: xchange-ftx/src/main/java/org/knowm/xchange/ftx/FtxAdapters.java [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+            ftxMarketDto -> {
+              CurrencyPairMetaData currencyPairMetaData =
+                  new CurrencyPairMetaData.Builder()
+                      .amountStepSize(ftxMarketDto.getPriceIncrement())
+                      .minimumAmount(ftxMarketDto.getSizeIncrement())
+                      .priceScale(ftxMarketDto.getPriceIncrement().scale())
+                      .baseScale(ftxMarketDto.getSizeIncrement().scale())
+                      .build();
+
+              if (ftxMarketDto.getType().equals("spot")) {
```
Comment (earce) [inline xchange-ftx/src/main/java/org/knowm/xchange/ftx/FtxAdapters.java:146]: move string to left side for NPE protection

File: xchange-ftx/src/main/java/org/knowm/xchange/ftx/FtxExchange.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import org.knowm.xchange.ftx.service.*;
+import org.knowm.xchange.service.account.AccountService;
+import org.knowm.xchange.service.marketdata.MarketDataService;
+import org.knowm.xchange.service.trade.TradeService;
+import org.knowm.xchange.utils.nonce.CurrentTimeIncrementalNonceFactory;
+import si.mazi.rescu.SynchronizedValueFactory;
+
+public class FtxExchange extends BaseExchange implements Exchange {
+
+  private final SynchronizedValueFactory<Long> nonceFactory =
```
Comment (earce) [inline xchange-ftx/src/main/java/org/knowm/xchange/ftx/FtxExchange.java:19]: This and the overridden method isn't necessary as it's the default.

File: xchange-core/src/main/java/org/knowm/xchange/dto/Order.java (hunk 1 above)
Comment (makarid) [reply xchange-core/src/main/java/org/knowm/xchange/dto/Order.java:437]: Hello, these are the exact words that FTX sends with their response. They have only these 3

**Pull request merged**